### PR TITLE
Feature/data sharing split UI admin researcher

### PR DESCRIPTION
**Pull request by andrewpatto**
As discussed - started the UI split of the data sharing into a "custodian" side and a "researcher" side.

This is not finished - but is already massive - so won't do the cleanups of each individual tab/form in this PR.

### CODE REVIEW

File: application/backend/dbschema/release.esdl [hunk 1]
```diff
@@ -218,7 +231,28 @@ module release {
...
         required property fileUrl -> str;
         required property fileSize -> int64;
-
     }
 
+    # A collection of settings for how data can be shared - including potentially
+    # bespoke fields for each sharing type (these fields only need to make sense once
+    # their particular sharing type is enabled - they will otherwise be blank or null or random).
+    #
+    type DataSharingConfiguration {
```
Comment (andrewpatto) [inline application/backend/dbschema/release.esdl:240]: a new object for grouping all the data sharing settings - both researcher and custodian edited.

File: application/backend/src/app.ts [hunk 2]
```diff
@@ -41,11 +41,17 @@ export class App {
...
+   * @param settings
+   * @param logger
+   * @param features
    */
   constructor(
     private readonly dc: DependencyContainer,
     private readonly settings: ElsaSettings,
-    private readonly logger: Logger
+    private readonly logger: Logger,
+    private readonly features: ReadonlySet<string>
```
Comment (andrewpatto) [inline application/backend/src/app.ts:54]: added the ability to configure a set of enabled features that propogate throughout the backend and frontend

File: application/backend/src/test-data/insert-test-data-release1.ts [hunk 3]
```diff
@@ -118,95 +125,21 @@ Ethics form XYZ.
...
+          whoId: "system",
+          occurredDateTime: e.datetime_current(),
+        })
+      ),
       dataEgressRecord: await makeSyntheticDataEgressRecord(),
     })
     .run(edgeDbClient);
 }
 
-function makeSytheticAuditLog() {
```
Comment (andrewpatto) [inline application/backend/src/test-data/insert-test-data-release1.ts:127]: we have enough audit activity working that we don't need a synthetic log in our test data anymore

**Comment by andrewpatto** [comment]
The e2e tests seem to be failing - but with nothing specific - just seems to be timing out after 12 mins??

**Comment by andrewpatto** [comment]
The custodians have the ability to enable/disable mechanisms and do control operations.

They can operate this panel even when the release is activated.

Screenshot 2023-04-18 at 4 50 19 pm

**Comment by andrewpatto** [comment]
The researcher now can only see the access panels that are activated.

Screenshot 2023-04-18 at 4 50 45 pm

Some fields - such as the field shown here - are editable by the researcher - but also appear in the Custodian UI

**Comment by DoxasticFox** [comment]
>The e2e tests seem to be failing - but with nothing specific - just seems to be timing out after 12 mins??

I ran them locally. They timeout after waiting for 30 seconds on my machine. The manual release dropdown never finishes loading:

Screen Shot 2023-04-18 at 4 59 24 pm

```
  1) [chromium] › dac-manual-release.spec.ts:3:1 › A release can be created manually which has selectable cases

    Test timeout of 30000ms exceeded.

    locator.click: Target closed
    =========================== logs ===========================
    waiting for locator('#datasetUriSelector > div').getByText('urn:fdc:umccr.org:2022:dataset/10c')
    ============================================================

      23 |     .locator("#datasetUriSelector > div")
      24 |     .getByText("urn:fdc:umccr.org:2022:dataset/10c")
    > 25 |     .click();
         |      ^
      26 |
      27 |   await page
      28 |     .locator("#datasetUriSelector > div")

        at /Users/chrnas/elsa-data/e2e/tests/dac-manual-release.spec.ts:25:6
```

**Review by williamputraintan — APPROVED** [review]

File: application/frontend/src/pages/releases/detail/access-box/object-signing-form.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  releaseKey: string;
+  releaseData: ReleaseTypeLocal;
+};
+
+type TsvColumnCheckProps = {
+  field: string;
+};
+
+const TsvColumnCheck: React.FC<TsvColumnCheckProps> = ({ field }) => (
+  <div key={field} className="flex items-center gap-2">
```
Comment (DoxasticFox) [inline application/frontend/src/pages/releases/detail/access-box/object-signing-form.tsx]: You copied my code from `manifest-form.tsx`, and my code is bad :sweat_smile: 

```suggestion
  <div className="flex items-center gap-2">
```

File: application/frontend/src/pages/releases/detail/access-box/gcp-storage-iam-share-form.tsx [hunk 5]
```diff
@@ -46,6 +46,9 @@ export const GcpStorageIamShareForm: React.FC<Props> = ({ releaseKey }) => {
 
   const iamUsersArray = iamUsers.split(/[ ,]+/).filter((u) => u);
 
+  const isSuccessState = (status: any): status is SuccessState => true;
+  const isErrorState = (status: any): status is ErrorState => true;
```
Comment (DoxasticFox) [inline application/frontend/src/pages/releases/detail/access-box/gcp-storage-iam-share-form.tsx:50]: I like the type guards! Making them more type-safe is only marginally more work:

```suggestion
  const isSuccessState = (status: any): status is SuccessState => 'recordsUpdated' in status;
  const isErrorState = (status: any): status is ErrorState => 'error' in status;
```

(Also, I think the "Something went wrong fetching release data" error will always be displayed even when there is no error unless you make them more type-safe).

**Review by DoxasticFox — APPROVED** [review]

File: application/frontend/src/pages/releases/detail/access-box/gcp-storage-iam-share-form.tsx (hunk 5 above)
Comment (andrewpatto) [reply application/frontend/src/pages/releases/detail/access-box/gcp-storage-iam-share-form.tsx:50]: Thanks! Yes - was just fixing the compile errors to get it going - we will need to refactor this form anyhow (we have to work out which bits of functionality move into the control panel and which stays)

**Pull request merged**